enhancements/monitoring/crd-based-cmo.md

```diff
@@ -0,0 +1,153 @@
+
+
+## Motivation
+
+* The specification is well known and to a degree self-documenting
+* The APIServer will validate user resources based on our specifications, so users get immediate feedback on errors instead of having to check if their config was applied and check logs.
+* Many users expect to interact with operators through a CRD
+* Compatible with GitOps workflows. 
+* We can add [cross]validation rules to CRD fields to avoid misconfigurations
+* End users get a much faster feedback loop. No more applying the config and scanning logs if things don't look right. The API server will give immediate feedback
+* Organizational users (such as ACM) can manage a single resource and observe its status
+
+
+### Goals
+
+- Replace configmaps with CRD
+- Smooth transition for users
+
+## Proposal
+
+## Design Details
+
+To initiate the process, let's establish a feature gate that will serve as the entry point for implementing a CRD configuration approach. This strategy enables us to make incremental advancements without the immediate burden of achieving complete feature equivalence with the config map. We can commence with a the basics and progressively incorporate additional functionalities as they develop.
+
+One proposal for a minimal DoD was:
+    - Feature gate
+    - CRD Initial dev https://github.com/openshift/cluster-monitoring-operator/pull/2347/
+        Add controller-gen logic to makefile
+        Add API to pkg/apis/cmo/v1
+        Add Generated CRD: config/crd/bases/example.com_clustermonitoringoperators.yaml
+        Add example CustomResource: config/examples/clustermonitoringoperator.yaml
+    - Client codegen: https://github.com/openshift/cluster-monitoring-operator/pull/2369
+    - Reconcile logic: https://github.com/openshift/cluster-monitoring-operator/pull/2350
+    - Add decoupling Confimgap / CustomResource:
+        Controller logic is strongly dependant of *manifests.Config struct.
+        
+
+
+### Overview
+
+- Replace confimgaps with CRD:
+  cluster-monitoring-configmap
+  user-workload-monitoring-config
+
+
+### Migration path
+
+    Feature gate. 
+    Switch mecanishm?
+    What to do if there are both CRD and confimgap?
+    Precedence CRD over configmap?
+    Should we compare CRD and configmap to check differences?
+
+### Issues
+
+- Decoupling Confimgap / CustomResource:
+        Controller logic is strongly dependant of *manifests.Config struct.
+        Should we translate CR into confimap?
+    
+- Correct name for apiVersion? In monitoring.coreos.com/v1 are all prometheus operator components, should we create a new one?
+
+- Refactor and clean up operator.go client.go manifests.config 
+
+
+### Transition to the user
+
+- How the user could adopt CR instead of configmap.
+
+### Example configuration
+
+
+#### CRD
+
+apiVersion: cmo.example.com/v1
+kind: ClusterMonitoring
```

Comment:
I think best naem for CRD is: Monitoring as it will be inside config,
What do you think @simonpasquier 
https://github.com/openshift/api/pull/1930#discussion_r1642601710